### PR TITLE
libpcre2: bump version

### PR DESCRIPTION
**Pull request by kenmays**
- Bumped to libpcre2 v10.43
- Readded: libpcre2 v10.39 for x86_gcc2 (GCC2) only.
- Builds tested on Haiku R1B4.

### CODE REVIEW

File: dev-libs/libpcre2/patches/libpcre2-10.43.patchset [hunk 1]
```diff
@@ -0,0 +1,54 @@
+From 375dd8b61787952e1bf23916567a3376a62b951c Mon Sep 17 00:00:00 2001
+From: Jerome Duval <jerome.duval@gmail.com>
+Date: Tue, 30 Apr 2019 18:33:26 +0200
+Subject: gcc2 fix
```
Comment (jmairboeck) [inline dev-libs/libpcre2/patches/libpcre2-10.43.patchset:4]: If this version isn't built with gcc2, it doesn't need fixes for it. Can't the newer version be also used for gcc2 then?

File: dev-libs/libpcre2/libpcre2-10.39.recipe
Comment (jmairboeck) [inline dev-libs/libpcre2/libpcre2-10.39.recipe:1]: This is a downgrade for x86_gcc2. Why not keep at least the current version 10.42?

Comment (kenmays) [reply dev-libs/libpcre2/libpcre2-10.39.recipe:1]: x86_gcc2 is currently at v10.39. No change for the moment. Just putting the recipe back in place.

File: dev-libs/libpcre2/patches/libpcre2-10.43.patchset (hunk 1 above)
Comment (kenmays) [reply dev-libs/libpcre2/patches/libpcre2-10.43.patchset]: Fixed.

**Comment by waddlesplash** [comment]
Why not just fix things for GCC2?

**Comment by Begasus** [comment]
For now re-enable previous version for gcc2. Adding an issue for newest being broken for gcc2.

**Comment by threedeyes** [comment]
After the update libpcre2, Qt applications stop working under 32 bit Haiku. The error is invalid opcode exception. If you roll back libpcre2 version, the problem disappears.

**Pull request merged**